src/coreclr/src/vm/i386/excepx86.cpp

```diff
@@ -1868,8 +1868,10 @@ NOINLINE LPVOID COMPlusEndCatchWorker(Thread * pThread)
     // Sync managed exception state, for the managed thread, based upon any active exception tracker
     pThread->SyncManagedExceptionState(fIsDebuggerHelperThread);
 
-    if (InlinedCallFrame::FrameHasActiveCall(pThread->m_pFrame))
+    if ((pThread->m_pFrame < esp) && InlinedCallFrame::FrameHasActiveCall(pThread->m_pFrame))
     {
+        // The current explicit frame is below the resume esp and it is an inlined call frame with an active call.
+        //
```

Comment:
Nit: You can fix typos in the comment below while you are on it:

unlined 
pinovoke